ERCS/erc-7832.md

```diff
@@ -0,0 +1,355 @@
+- **Dynamic Minting Fees**: To align token costs with user activity and ownership.  
+- **Donation-Based Engagement**: Encouraging contributions to creators.  
+- **Role-Based Access**: Empowering creators while ensuring transparent governance.  
+
+
+## Specification  
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+---
+
+1. **Variables and Constants**
+    #### `mintBaseFee`
+    **Type**: `uint256`  
+    **Description**:  
+    The base fee for minting a token, paid by the user to create a new token.
+
+    #### `creatorSignatureFee`
+    **Type**: `uint256`  
+    **Description**:  
+    The fee required for a user to acquire a creator's signature, allowing them to become a creator in the system.
+
+    #### `maxMintsPerUserInCycle`
+    **Type**: `uint256`  
+    **Description**:  
+    The maximum number of mints a user can perform in a cycle. Once the limit is exceeded, the user's minting count is reset.
+
+    #### `lastUpdateTimestamp`
+    **Type**: `uint256`  
+    **Description**:  
+    Timestamp of the last time the contract terms were updated (e.g., minting fees and creator signature fees). It is used to determine when the contract's terms can be updated again.
+
+    #### `UPDATE_INTERVAL`
+    **Type**: `uint256` (constant)  
+    **Description**:  
+    The time interval between contract terms updates. **MUST** be fixed to <ins>30 days</ins>.
+
+    #### `CREATOR_ROLE`
+    **Type**: `bytes32` (constant)  
+    **Description**:  
+    The role identifier for creators in the system. 
+
+    #### `CONTRIBUTOR_ROLE`
+    **Type**: `bytes32` (constant)  
+    **Description**:  
+    The role identifier for contributors in the system. 
+
+    #### `mintsPerUserInCycle`
+    **Type**: `mapping(address => uint256)`  
+    **Description**:  
+    A mapping that tracks the number of mints a user has performed in the current cycle. It is used to enforce the maximum minting limit per user.
+
+    ---
+
+2. **Events**
+    #### `CreatorTermsUpdated(uint256 mintBaseFee, uint256 creatorSignatureFee, uint256 maxMintsPerUserInCycle)`
+    **Description**:  
+    Emitted when the contract terms related to minting are updated by the `DEFAULT_ADMIN_ROLE`.
+
+
+    #### `DonationReceived(address from, address to, uint256 amount)`
+    **Description**:  
+    Emitted when a user donates ETH to a creator. This event tracks the details of the donation, including the donor's address, the recipient's address, and the donation amount.
+
+    **Parameters**:  
+    - `from`: The address of the user making the donation.  
+    - `to`: The address of the creator receiving the donation.  
+    - `amount`: The amount of ETH donated.
+
+
+    ---
+
+3. **Modifiers**
+    #### `onlyIfNotPaused`
+    **Description**:  
+    Ensures that the contract **MUST NOT** be paused when executing a function.
+
+    ---
```

Comment:
```suggestion
```

Modifiers are only one possible way to implement a particular externally visible behaviour. Instead of defining _how_ the contract implements something, only specify _what_ the contract has to enforce.